formats/proto/src/main/java/io/cloudevents/proto/ProtoCloudEvent.java

```diff
@@ -0,0 +1,392 @@
+package io.cloudevents.proto;
+
+import com.google.protobuf.Message;
+import com.google.protobuf.Timestamp;
+import io.cloudevents.CloudEventData;
+import io.cloudevents.SpecVersion;
+import io.cloudevents.v1.proto.CloudEvent;
+
+import javax.json.Json;
+import javax.json.JsonReader;
+import javax.json.JsonStructure;
+import javax.json.JsonValue;
+import java.io.StringReader;
+import java.net.URI;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Base64;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * A CloudEvent that is backed by a Google Protobuf structure.
+ */
+class ProtoCloudEvent implements io.cloudevents.CloudEvent {
```

Comment:
I would love to avoid reimplementing `CloudEvent`, i prefer the #348 that copies values from the proto cloudevent

Reply:
I did think abut that, my concern was really from an efficiency standpoint to avoid copying values around when potentially the SKD is being used in the context of a router where maybe there's just a need to access the 'type' (for example) to select a transport/endpoint to send it to.

If the preferred model is to always clone the values I can certainly change to that for now to get stuff going but it might be worth doing some profiling to see if an optimization might be appropriate in the future.